### PR TITLE
EZP-27057: Section details in details tab

### PR DESCRIPTION
**Pull request by StephaneDiot**
JIRA: https://jira.ez.no/browse/EZP-27057

## Description

Be able to view section details in detail tab.

## Screencast

https://youtu.be/7z6SjTz7M-4
## Tests

Unit & manually tested.

### CODE REVIEW

**Comment by StephaneDiot** [comment]
ping @dpobel  for quick review before I start unit tests

File: Resources/public/js/models/ez-sectionmodel.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-sectionmodel', function (Y) {
+    "use strict";
+    /**
+     * Provides the Content model class
```
Comment (yannickroger) [inline Resources/public/js/models/ez-sectionmodel.js]: section

File: Resources/public/js/views/services/ez-locationviewviewservice.js [hunk 2]
```diff
@@ -468,9 +468,22 @@ YUI.add('ez-locationviewviewservice', function (Y) {
...
+                                return;
+                            }
+
+                            service.set('section', section);
+                            endSectionLoad();
+                        });
                     });
                 });
-
+
```
Comment (yannickroger) [inline Resources/public/js/views/services/ez-locationviewviewservice.js]: useless

**Comment by StephaneDiot** [comment]
Ready for review @dpobel @yannickroger  :crossed_fingers:

File: Resources/translations/locationview.en.xlf [hunk 3]
```diff
@@ -234,6 +240,30 @@
         <note>key: locationview.details.retry</note>
         <jms:reference-file>Resources/public/templates/tabs/details.hbt</jms:reference-file>
       </trans-unit>
+      <trans-unit id="331cffcd3a384170271d5718f985aceb785c1c66" resname="locationview.details.sectionId">
+        <source>Section id</source>
+        <target>Section id</target>
```
Comment (dpobel) [inline Resources/translations/locationview.en.xlf]: according to the specs, it is supposed to be `Section ID`

File: Resources/translations/locationview.en.xlf [hunk 4]
```diff
@@ -234,6 +240,30 @@
...
+      </trans-unit>
+      <trans-unit id="a7269661e9fb6263e1384653afec0a71a486f464" resname="locationview.details.sectionIdentifier">
+        <source>Section Identifier</source>
+        <target>Section Identifier</target>
+        <note>key: locationview.details.sectionIdentifier</note>
+        <jms:reference-file>Resources/public/templates/tabs/details.hbt</jms:reference-file>
+      </trans-unit>
+      <trans-unit id="3856df0053db711c29636338a59538cba6951220" resname="locationview.details.sectionName">
+        <source>Section name</source>
+        <target>Section name</target>
```
Comment (dpobel) [inline Resources/translations/locationview.en.xlf]: according to the specs, it is supposed to be `Section Name`

File: Resources/translations/locationview.en.xlf [hunk 5]
```diff
@@ -234,6 +240,30 @@
...
+        <note>key: locationview.details.sectionName</note>
+        <jms:reference-file>Resources/public/templates/tabs/details.hbt</jms:reference-file>
+      </trans-unit>
+      <trans-unit id="eca9605beb0cee0f9908fcb213519c57b987bf80" resname="locationview.details.sectiondetails">
+        <source>Section details</source>
+        <target>Section details</target>
```
Comment (dpobel) [inline Resources/translations/locationview.en.xlf]: according to the specs, it is supposed to be `Section Details`

**Comment by StephaneDiot** [comment]
updated @dpobel @yannickroger

File: Resources/public/js/views/services/plugins/ez-sectionloadplugin.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+     * @module ez-sectionloadplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Object section load plugin.
+     *
+     * In order to use it you need to fire the `loadSection` event with two parameters:
+     *  - `content` the content containing the section to be loaded
+     *  - `callback` the callback called in case of error
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-sectionloadplugin.js]: actually always called

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 7]
```diff
@@ -182,6 +191,31 @@ YUI.add('ez-locationviewdetailstabview', function (Y) {
...
+         * @protected
+         */
+        _fireLoadSection: function () {
+            var callback = Y.bind(function (error, section) {
+                    if (error) {
+                        this.fire('error', Y.eZ.trans('failed.to.load.section', {}, 'locationview'));    
+                    } else {
+                        this.set('section', section);
+                    }
+            }, this);
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: wrong indent here

File: Resources/public/js/views/tabs/ez-locationviewdetailstabview.js [hunk 8]
```diff
@@ -182,6 +191,31 @@ YUI.add('ez-locationviewdetailstabview', function (Y) {
...
+            /**
+             * Fired when the details view needs section
+             * @event loadSection
+             * @param {eZ.Content} content the content
+             * @param {Function} errorCallback the function called in case of section load error.
```
Comment (dpobel) [inline Resources/public/js/views/tabs/ez-locationviewdetailstabview.js]: not `errorCallback`, but `callback` according to line 216

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 9]
```diff
@@ -584,10 +598,69 @@ YUI.add('ez-locationviewdetailstabview-tests', function (Y) {
...
+
+            this.view.render();
+            this.view.set('active', true);
+            this.view.set('loadingError', true);
+
+            this.view.on('loadSection', function () {
+                loadSection = true;
+            });
+
+            this.view.get('container').one('.ez-asynchronousview-retry').simulateGesture('tap', this.next(Y.bind(function () {
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: `Y.bind` call is useless, the second parameter of `next` is the context in which the function will be executed.

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 10]
```diff
@@ -584,10 +598,69 @@ YUI.add('ez-locationviewdetailstabview-tests', function (Y) {
...
+                    Y.Assert.isUndefined(
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: indent...

File: Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js [hunk 11]
```diff
@@ -584,10 +598,69 @@ YUI.add('ez-locationviewdetailstabview-tests', function (Y) {
...
+            _eventTestSetup(this);
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+            delete this.view;
+        },
+
+        "Should fire the loadSection event when view become active": function () {
+            var that = this,
```
Comment (dpobel) [inline Tests/js/views/tabs/assets/ez-locationviewdetailstabview-tests.js]: please bind

File: Tests/js/models/assets/ez-contentmodel-tests.js [hunk 12]
```diff
@@ -886,6 +890,84 @@ YUI.add('ez-contentmodel-tests', function (Y) {
...
+                callbackCalled = false,
+                error = false;
+            
+            Y.eZ.Section = this._getSectionModel(error, options);
+
+            this.model.loadSection(options, Y.bind(function (err, section) {
+                callbackCalled = true;
+
+                Assert.isFalse(err, 'error should be false');
+                Assert.isTrue(section instanceof Y.eZ.Section, 'Should return the section');
```
Comment (dpobel) [inline Tests/js/models/assets/ez-contentmodel-tests.js]: https://yuilibrary.com/yui/docs/api/classes/Test.Assert.html#method_isInstanceOf

File: Resources/public/js/views/services/plugins/ez-sectionloadplugin.js [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+         * Loads a Section. Once this is done, it sets the section in
+         * the attribute defined in the event parameter of the event target.
+         * @protected
+         * @method _loadSection
+         * @param {Object} e loadSection event facade
+         */
+        _loadSection: function (e) {
+            var loadOptions = {api: this.get('host').get('capi')};
+            
+            e.content.loadSection(loadOptions, function(error, section) {
```
Comment (dpobel) [inline Resources/public/js/views/services/plugins/ez-sectionloadplugin.js]: can be simplified to 
```js
e.content.loadSection(loadOptions, e.callback);
```

File: Resources/public/js/models/ez-contentmodel.js [hunk 14]
```diff
@@ -355,7 +355,27 @@ YUI.add('ez-contentmodel', function (Y) {
...
+            var section = new Y.eZ.Section();
 
+            section.set('id', this.get('resources').Section);
+            section.load(options, function(error) {
+                if (error) {
+                    callback(error);
+                } else {
+                    callback(error, section);
+                }
+
```
Comment (yannickroger) [inline Resources/public/js/models/ez-contentmodel.js]: nitpick: empty line

**Pull request merged**